src/clr-angular/forms/select/providers/option-selection.service.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import {Injectable} from "@angular/core";
+import {Subject} from "rxjs";
+import {Observable} from "rxjs/internal/Observable";
+
+import {ClrOption} from "../option";
+
+@Injectable()
+export class OptionSelectionService {
+    // Array as we need support for selecting multiple options
+    currentSelection: Array<ClrOption> = [];
+
+    private _selectionChanged: Subject<void> = new Subject<void>();
+
+    get selectionChanged(): Observable<void> {
+        return this._selectionChanged.asObservable();
+    }
+
+    // Currently only handles single selection
+    updateSelection(option: ClrOption): void {
+        if (this.currentSelection.indexOf(option) > -1) {
```

Comment:
I'm struggling with this method, its written to support an array of selected items but then forced to be only single selection. I'd like this method (and service) to be either 1) written to just handle single selection or 2) written to be ready for multi-selection.

Reply:
I understand the confusion. I started out with thinking that I can code for both multi and single select right now but then realized that clicking on an option again in multi selection toggles the selection, unlike in single selection. For now, I will just code it for single selection and we can handle multi selection after MVP.